addons/event_sale/models/event_registration.py

```diff
@@ -30,7 +30,8 @@ def _compute_registration_status(self):
             cancelled_so_registrations = registrations.filtered(lambda reg: reg.sale_order_id.state == 'cancel')
             cancelled_so_registrations.state = 'cancel'
             cancelled_registrations = cancelled_so_registrations | registrations.filtered(lambda reg: reg.state == 'cancel')
-            if not so_line or float_is_zero(so_line.price_total, precision_rounding=so_line.currency_id.rounding):
+            if not so_line or (float_is_zero(so_line.price_total, precision_rounding=so_line.currency_id.rounding)
+                               and so_line.order_id.amount_total == 0):
```

Comment:
I guess we can just `float_is_zero(so_line.order_id.amount_total, precision_rounding=so_line.order_id.currency_id.rounding)`